extensions/sinks/kafka/ext/kafka.go

```diff
@@ -243,7 +243,7 @@ func (m *kafkaSink) Collect(ctx api.StreamContext, item interface{}) error {
 				continue
 
 			case kafkago.Error:
-				if err.Temporary() {
+				if err.Temporary() || strings.Contains(err.Error(), "kafka.(*Client).Produce:") {
```

Comment:
I don't think this works, and this is irrelevant to the host not found problem. It is not a kafkago.Error actually.